raiden/utils/upgrades.py

```diff
@@ -119,27 +106,27 @@ def run(self):
 
             self._copy(str(old_db_filename), str(self._current_db_filename))
 
-            storage = SQLiteStorage(str(self._current_db_filename), JSONSerializer())
+            storage = SQLiteStorage(str(self._current_db_filename))
 
             log.debug(f'Upgrading database from {older_version} to v{RAIDEN_DB_VERSION}')
 
-            cursor = storage.conn.cursor()
-            with in_transaction(cursor):
-                try:
+            try:
+                with storage.transaction():
                     version_iteration = older_version
                     for upgrade_func in UPGRADES_LIST:
                         version_iteration = self._run_upgrade_func(
-                            cursor,
+                            storage,
                             upgrade_func,
                             version_iteration,
                         )
 
-                    update_version(cursor, RAIDEN_DB_VERSION)
+                    update_version(storage, RAIDEN_DB_VERSION)
                     # Prevent the upgrade from happening on next restart
                     self._backup_old_db(old_db_filename)
-                except RaidenDBUpgradeError:
-                    self._delete_current_db()
-                    raise
+            except Exception as e:
```

Comment:
@rakanalh Why did you remove the specific `RaidenDBUpgradeError` exception catching from here and put a catch-all exception? Catch-alls are something to be avoided.

Reply:
In the case of migrations, we agreed that we wanted to catch all exceptions so that proper rollback handling is done in case of ANY failure.

Reply:
The class at that point didn't really make much more sense so i removed it